linux/buildSrc/src/main/java/net/adoptopenjdk/installer/UploadDebianPackage.java

```diff
@@ -46,10 +57,44 @@ protected void uploadPackages(ArtifactoryCredentials artifactoryCredentials) {
                 getArchitecture()
         );
 
-        executor.submit(UploadPackageTask.class, workerConfiguration -> {
-            workerConfiguration.setIsolationMode(IsolationMode.NONE);
-            workerConfiguration.setParams(getApiEndpoint(), artifactoryCredentials, upload);
-        });
+        this.getArtifactoryClient()
+                .repository(upload.getRepository())
+                .upload(upload.getRemotePath(), upload.getFile())
+                .withProperty("deb.distribution", upload.getDistributions().toArray())
+                .withProperty("deb.component", upload.getComponents().toArray())
+                .withProperty("deb.architecture", upload.getArchitectures().toArray())
+                .bySha1Checksum()
+                .doUpload();
+    }
+
+    private boolean artifactExists() {
+        List<RepoPath> searchResults = this.getArtifactoryClient()
+                .searches()
+                .repositories(getRepository())
+                .artifactsByName(getPackageToPublish().getName())
+                .doSearch();
+
+        if (searchResults.isEmpty()) {
+            return false;
+        }
+
+        for (RepoPath searchResult : searchResults) {
+            if (Objects.equals(getRepository(), searchResult.getRepoKey())
+                    && Objects.equals(this.targetFilePath(), searchResult.getItemPath())) {
+                return true;
+            }
+        }
+
+        return false;
```

Comment:
Should we log this case or is it common?

Reply:
It‘s logged, line 31.